lib/data/clientToolsCache.ts

```diff
@@ -0,0 +1,156 @@
+            description: tool.description || "",
+            parameters: tool.parameters || {},
+            serverRequestUrl: tool.serverRequestUrl!,
+            updatedAt: new Date(),
+          })
+          .where(eq(clientToolsCache.id, existingTool.id));
+      } else {
+        // Insert new record
+        await db.insert(clientToolsCache).values({
+          toolName,
+          description: tool.description || "",
+          parameters: tool.parameters || {},
+          serverRequestUrl: tool.serverRequestUrl!,
+          customerEmail: customerEmail ?? null,
+          platformCustomerId: platformCustomerId ?? null,
+          unused_mailboxId: 0,
+        });
+      }
+    } catch (error) {
+      captureExceptionAndLog(error);
+      throw error;
+    }
+  }
+};
+
+export const getCachedClientTools = async ({
+  customerEmail,
+  platformCustomerId,
+}: GetCachedClientToolsParams): Promise<Record<string, ToolRequestBody>> => {
+  const scopedFilters = [eq(clientToolsCache.unused_mailboxId, 0)];
+  if (customerEmail) scopedFilters.push(eq(clientToolsCache.customerEmail, customerEmail));
+  if (platformCustomerId) scopedFilters.push(eq(clientToolsCache.platformCustomerId, platformCustomerId));
+
+  let cachedTools =
+    customerEmail || platformCustomerId
+      ? await db.query.clientToolsCache.findMany({
+          where: and(...scopedFilters),
+        })
+      : [];
+
+  if (cachedTools.length === 0) {
+    cachedTools = await db.query.clientToolsCache.findMany({
+      where: and(
+        eq(clientToolsCache.unused_mailboxId, 0),
+        isNull(clientToolsCache.customerEmail),
+        isNull(clientToolsCache.platformCustomerId),
+      ),
+    });
+  }
+
+  return Object.fromEntries(
+    cachedTools.map((tool) => [
+      tool.toolName,
+      {
+        description: tool.description,
+        parameters: tool.parameters,
+        serverRequestUrl: tool.serverRequestUrl,
+      } satisfies ToolRequestBody,
+    ]),
+  );
+};
+
+export const clearCachedClientTools = async ({
+  customerEmail,
+  platformCustomerId,
+}: GetCachedClientToolsParams): Promise<void> => {
+  const where = [eq(clientToolsCache.unused_mailboxId, 0)];
+  if (customerEmail) where.push(eq(clientToolsCache.customerEmail, customerEmail));
+  if (platformCustomerId) where.push(eq(clientToolsCache.platformCustomerId, platformCustomerId));
+
+  // If no identifiers provided, target global entries
+  if (where.length === 1) {
+    where.push(isNull(clientToolsCache.customerEmail), isNull(clientToolsCache.platformCustomerId));
+  }
+
+  await db.delete(clientToolsCache).where(and(...where));
+};
+
+export const getCachedToolsForAdmin = async (
+  tx: Transaction | typeof db = db,
+): Promise<
+  {
+    toolName: string;
+    description: string;
+    parameters: Record<string, any>;
+    serverRequestUrl: string;
+    customerEmail: string | null;
+    platformCustomerId: string | null;
+    createdAt: Date;
+    updatedAt: Date;
+  }[]
+> => {
```

Comment:
```suggestion
) => {
```

I don't think we need an explicit return type